### DISCUSSION TITLE
Cool Project 👌

### DISCUSSION THREAD

**Discussion by prologic** [alda-lang/alda#494] [Ideas]
Q: Have you thought about rewriting the player in Go too? 🤔

**Comment by daveyarwood** [comment]
Thanks!

I chose Kotlin for the player because it's a JVM language, and the JVM has a built-in MIDI sequencer and synthesizer that works out of the box on every operating system. That's nice because it means that users don't need to install anything extra in order to use Alda.

**Reply by prologic** [reply]
Wow! I had no idea the JVM had a builtin MIDI sequencer 😱

**Reply by daveyarwood** [reply]
Yeah, and it's actually pretty good! Surprising, right?

**Reply by daveyarwood** [reply]
More info here: https://docs.oracle.com/javase/8/docs/technotes/guides/sound/

**Reply by prologic** [reply]
Out of curiosity is there no decent MIDI sequencer in Go?

**Reply by daveyarwood** [reply]
Not that I'm aware of.
